src/api/entities/Asset/NonFungible/Nft.ts

```diff
@@ -198,6 +208,71 @@ export class Nft extends Entity<NftUniqueIdentifiers, HumanReadable> {
     return null;
   }
 
+  /**
+   * Get owner of the NFT
+   *
+   * @note This method returns `null` if there is no existing holder for the token. This may happen even if the token has been redeemed/burned
+   */
+  public async getOwner(): Promise<DefaultPortfolio | NumberedPortfolio | null> {
+    const {
+      collection: { ticker },
+      id,
+      context: {
+        polymeshApi: {
+          query: {
+            nft: { nftOwner },
+          },
+        },
+      },
+      context,
+    } = this;
+
+    const rawTicker = stringToTicker(ticker, context);
+    const rawNftId = bigNumberToU64(id, context);
+
+    const owner = await nftOwner(rawTicker, rawNftId);
+
+    if (owner.isEmpty) {
+      return null;
+    }
+
+    return meshPortfolioIdToPortfolio(owner.unwrap(), context);
+  }
+
+  /**
+   * Check if the NFT is locked in any settlement instruction
+   *
+   * @throws if NFT has no owner (has been redeemed)
+   */
+  public async isLocked(): Promise<boolean> {
+    const {
+      collection: { ticker },
+      id,
+      context: {
+        polymeshApi: {
+          query: { portfolio },
+        },
+      },
+      context,
+    } = this;
+
+    const owner = await this.getOwner();
```

Comment:
I think this check should be encapsulated in the `exists` method, and here just check that method

Reply:
We need the owner value to check locked status